### PR TITLE
Read commited

### PR DESCRIPTION
**Pull request by hari-om-888**

### CODE REVIEW

File: src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java [hunk 1]
```diff
@@ -197,6 +196,15 @@ private TransactionClose(String msg) {
         return result;
     }
 
+    private <T> T getTransaction(IsolationLevel isolationLevel, Transaction tx) {
+        if(IsolationLevel.READ_COMMITTED_NO_WRITE.equals(isolationLevel)
```
Comment (hari-om-888) [inline src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java:190]: https://apple.github.io/foundationdb/developer-guide.html#snapshot-reads
With snapshot reads, reads will see last successfully committed transaction version of the key. The key values may change after the read occurred.

File: src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java [hunk 2]
```diff
@@ -154,7 +151,8 @@ private TransactionClose(String msg) {
         byte[] value = null;
         for (int i = 0; i < maxRuns; i++) {
             try {
-                value = this.tx.get(key).get();
+                ReadTransaction transaction  = getTransaction(this.isolationLevel, this.tx);
```
Comment (hari-om-888) [inline src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java:156]: moving reads to another method to leverage snapshot reads.
https://apple.github.io/foundationdb/developer-guide.html#snapshot-reads

File: src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java [hunk 3]
```diff
@@ -178,7 +176,8 @@ private TransactionClose(String msg) {
         for (int i = 0; i < maxRuns; i++) {
             final int startTxId = txCtr.get();
             try {
-                result = tx.getRange(new Range(startKey, endKey), limit).asList().get();
+                ReadTransaction transaction = getTransaction(isolationLevel, this.tx);
+                result = transaction.getRange(new Range(startKey, endKey), limit).asList().get();
```
Comment (hari-om-888) [inline src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java]: same as above

File: src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java [hunk 4]
```diff
@@ -1,9 +1,6 @@
 package com.experoinc.janusgraph.diskstorage.foundationdb;
 
-import com.apple.foundationdb.Database;
-import com.apple.foundationdb.KeyValue;
-import com.apple.foundationdb.Range;
-import com.apple.foundationdb.Transaction;
+import com.apple.foundationdb.*;
```
Comment (skyrocknroll) [inline src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java]: Good to move away from wildcard imports

**Review by skyrocknroll — CHANGES_REQUESTED** [review]

File: src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java (hunk 4 above)
Comment (hari-om-888) [reply src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java]: For not conflicting imports, it doesn't matter. Also these days, intellij does the optimization of imports. this was done by intellij, auto format indents.

Comment (skyrocknroll) [reply src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java]: I would still recommend using explicit import which will make the code more readable.  Once we start having multiple wildcard imports then the dance starts :) I would request to disable the IntelliJ feature which collapsed the explicit imports. Intellij is supposed to make the code more readable not the other way around :D

Comment (hari-om-888) [reply src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java]: Code Readability is not affected by imports. It is pretty easy to track which dependency is used. Also if some imports collide, intellij will not collapse imports to *
Anyways, updated the pr to move away from collapsed imports.

**Review by upaangsaxena — APPROVED** [review]

**Pull request closed without merging**